utils/gen_from_dfn.py

```diff
@@ -160,3 +166,47 @@ def render_template(template_name: str, output_path: str, **context):
         keywords=keywords,
         valids=valids,
     )
+
+    # Export hover data from dfn files
+    # common.dfn is a special file that contains common descriptions for keywords
+    # which are used to replace placeholders in other dfn files
+    common = {}
+    for section in Dfn(Path("data/dfn/common.dfn")).data:
+        if not section.startswith("name"):
+            continue
+        name, description = section.strip().split("\n")
+        if not (name.startswith("name") and description.startswith("description")):
+            continue
+        common[name.split(maxsplit=1)[-1]] = description.split(maxsplit=1)[-1]
+
+    hover = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+    # Some dfn keywords have to be replaced with a different keyword from common.dfn
+    # e.g. see  gwe-ctp.dfn, "auxiliary" keyword is replaced with "auxnames" from common
+    keyword_alias = {"auxiliary": "auxnames", "print_stage": "print_head"}
+
+    for dfn_file in Path("data/dfn").glob("*.dfn"):
+        dfn = Dfn(dfn_file)
+        for section in dfn.sections:
+            if section.tagged and (description := section.description):
+                if "REPLACE" in description:  # Replace placeholders from common
+                    keyword = keyword_alias.get(section.keyword, section.keyword)
+                    description = common[keyword]
+                    # Create replacement dictionary from the description
+                    replacement = eval(section.description.strip(f"REPLACE {keyword} "))
+                    for key, value in replacement.items():
+                        description = description.replace(key, value)
```

Comment:
I think you should loop over the items to replace inside `description`. The description string may contain multiple instances of "{#1}".

Can you try using `str.format_map()`?